client/me/security/controller.js

```diff
@@ -62,9 +63,24 @@ export function accountRecovery( context, next ) {
 }
 
 export function socialLogin( context, next ) {
+	// Remove id_token from the address bar and push social connect args into the state instead
+	if ( context.hash && context.hash.client_id ) {
+		page.replace( context.path, context.hash );
+		return;
+	}
+
+	const previousHash = context.state || {};
+	const { client_id, user_email, user_name, id_token, state } = previousHash;
```

Comment:
doesn't seem necessary to destructure here

Reply:
Same reason I think it's a nice way to make sure our response has all properties, no more no less (also useful to document what we expect the `socialServiceResponse` format to be.